packages/vitest/src/node/config.ts

```diff
@@ -135,13 +135,21 @@ export function resolveConfig(
     }
   }
 
+  // TODO: V2.0.0 remove
   // @ts-expect-error -- check for removed API option
   if (resolved.coverage.provider === 'c8')
     throw new Error('"coverage.provider: c8" is not supported anymore. Use "coverage.provider: v8" instead')
 
   if (resolved.coverage.provider === 'v8' && resolved.coverage.enabled && isBrowserEnabled(resolved))
     throw new Error('@vitest/coverage-v8 does not work with --browser. Use @vitest/coverage-istanbul instead')
 
+  if (resolved.coverage.enabled && resolved.coverage.reportsDirectory) {
+    const reportsDirectory = resolve(resolved.root, resolved.coverage.reportsDirectory)
+
+    if (reportsDirectory === resolved.root || reportsDirectory === process.cwd())
```

Comment:
Maybe check also parent directory like `../`?
Using `root.startsWith(reportsDirectory)` can cover `root === reportsDirectory` as a special case.

Reply:
We can check `startsWith(reportsDirectory + '/')`

Reply:
Sounds good. We can also add this improvement later on if you want to include this in the release @sheremet-va.